parsers/US_PJM.py

```diff
@@ -65,332 +72,283 @@
     logger: Logger = getLogger(__name__),
-) -> list:
+) -> list[dict]:
     """Gets consumption forecast for specified zone."""
 
-    if target_datetime:
-        raise NotImplementedError("This parser is not yet able to parse past dates")
-    if not session:
-        session = Session()
+    if target_datetime is not None:
+        raise ParserException(
+            PARSER, "This parser is not yet able to parse historical data", zone_key
+        )
 
-    # startRow must be set if forecast_area is set.
-    # RTO_COMBINED is area for whole PJM zone.
+    session = session or Session()
+    # startRow must be set if forecast_area is set. RTO_COMBINED is area for whole PJM zone.
     params = {"download": True, "startRow": 1, "forecast_area": "RTO_COMBINED"}
+    data = _fetch_api_data(kind="load_frcstd_7_day", params=params, session=session)
 
-    # query API
-    data = fetch_api_data(kind="load_frcstd_7_day", params=params, session=session)
-
-    data_points = []
+    consumption_list = TotalConsumptionList(logger)
     for elem in data:
         utc_datetime = elem["forecast_datetime_beginning_utc"]
-        data_point = {
-            "zoneKey": zone_key,
-            "datetime": datetime.fromisoformat(utc_datetime).replace(
-                tzinfo=timezone.utc
-            ),
-            "value": elem["forecast_load_mw"],
-            "source": "pjm.com",
-        }
-        data_points.append(data_point)
-
-    return data_points
+        consumption_list.append(
+            zoneKey=zone_key,
+            datetime=datetime.fromisoformat(utc_datetime)
+            .replace(tzinfo=timezone.utc)
+            .astimezone(TIMEZONE),
+            source=SOURCE,
+            consumption=elem["forecast_load_mw"],
+            sourceType=EventSourceType.forecasted,
+        )
+    return consumption_list.to_list()
 
 
 @refetch_frequency(timedelta(days=1))
 def fetch_production(
-    zone_key: str = "US-PJM",
-    session: Session = Session(),
+    zone_key: ZoneKey = ZONE_KEY,
+    session: Session | None = None,
     target_datetime: datetime | None = None,
     logger: Logger = getLogger(__name__),
-) -> list:
-    """uses PJM API to get generation  by fuel. we assume that storage is battery storage (see https://learn.pjm.com/energy-innovations/energy-storage)"""
+) -> list[dict]:
+    """Uses PJM API to get generation by fuel.
+
+    We assume that storage is battery storage (see https://learn.pjm.com/energy-innovations/energy-storage)
+    """
     if target_datetime is None:
         target_datetime = datetime.now(timezone.utc)
 
+    session = session or Session()
+
     params = {
         "download": True,
         "startRow": 1,
-        "fields": "datetime_beginning_ept,fuel_type,mw",
-        "datetime_beginning_ept": target_datetime.strftime("%Y-%m-%dT%H:00:00.0000000"),
+        "fields": "datetime_beginning_utc,fuel_type,mw",
+        "datetime_beginning_utc": target_datetime.strftime("%Y-%m-%dT%H:00:00.0000000"),
```

Comment:
fix: the previous code was feeding a UTC timestamp into a field expecting [EPT timestamps](https://www.pjm.com/-/media/etools/exschedule/exschedule-dst-guide.ashx#:~:text=Eastern%20Prevailing%20Time%20(EPT)%20is,one%20hour%20to%203%3A00.)

> Eastern Prevailing Time (EPT) is the time on an automatically updated clock, such as on a cellphone or computer. Clocks change from Eastern Standard Time (EST) to Eastern Daylight Time (EDT) on March 10, 2024. At 2:00 EPT, clocks will “spring forward” one hour to 3:00. In other words, this hour from 2:00 to 02:59 “doesn’t exist”.

Luckily the API accepts datetimes as UTC through the `datetime_beginning_utc` so we can just use that without having to deal with timezone conversions

Reply:
question: should I cast `target_datetime.astimezone(timezone.utc)` just in case or do we assume inputs have always been validated to be UTC?